typing/printtyp.ml

```diff
@@ -2445,9 +2515,7 @@ let warn_on_missing_def env ppf t =
         fprintf ppf
           "@,@[<hov>Type %a is abstract because@ no corresponding\
            @ cmi file@ was found@ in path.@]" (Style.as_inline_code path) p
-    | {type_kind =
-       Type_abstract Abstract_def | Type_record _ | Type_variant _ | Type_open }
-      -> ()
+    | _ -> ()
```

Comment:
In principle, I would prefer it not to be robust, because it's entire purpose is to pick exactly the cases marked with this constructor, and this constructor's entire purpose is to be matched by this case. There is no reasonable way to update the type with new constructors that would require adding more cases to this match.

I'm fine to update it though on the basis of some "always use robust matches" rule of thumb that is applied even when it is not actually beneficial -- I can easily be convinced it is better to have a simple rule and apply it universally.

Reply:
I disagree with your analysis in the first paragraph. Currently `rec_check_regularity` is the only case of dummy names that we handle in this way, but I could very well see other type-checking processes injecting dummy abstract types for different reasons, and expecting to be asked to update the current function with a specialized error message.
(But: it may be that I don't understand the usage context for the somewhat arcane `warn_on_missing_def`, and that the above is unlikely due to a very restricted usage scenario.)